src/stream/src/executor/source/source_executor.rs

```diff
@@ -490,19 +505,72 @@ impl<S: StateStore> SourceExecutor<S> {
 
         let recover_state: ConnectorState = (!boot_state.is_empty()).then_some(boot_state);
         tracing::debug!(state = ?recover_state, "start with state");
-        let (source_chunk_reader, latest_splits) = self
-            .build_stream_source_reader(
-                &source_desc,
-                recover_state,
-                // For shared source, we start from latest and let the downstream SourceBackfillExecutors to read historical data.
-                // It's highly probable that the work of scanning historical data cannot be shared,
-                // so don't waste work on it.
-                // For more details, see https://github.com/risingwavelabs/risingwave/issues/16576#issuecomment-2095413297
-                // Note that shared CDC source is special. It already starts from latest.
-                self.is_shared_non_cdc && is_uninitialized,
+
+        let mut barrier_stream = barrier_to_message_stream(barrier_receiver).boxed();
+        let mut reader_and_splits: Option<(BoxChunkSourceStream, Option<Vec<SplitImpl>>)> = None;
+        let seek_to_latest = self.is_shared_non_cdc && is_uninitialized;
+        let source_reader = source_desc.source.clone();
+        let (column_ids, source_ctx) = self.prepare_source_stream_build(&source_desc);
+        let source_ctx = Arc::new(source_ctx);
+        let mut build_source_stream_fut = Box::pin(async move {
+            let backoff = get_infinite_backoff_strategy();
+            tokio_retry::Retry::spawn(backoff, || async {
+                match source_reader
+                    .build_stream(
+                        recover_state.clone(),
+                        column_ids.clone(),
+                        source_ctx.clone(),
+                        seek_to_latest,
+                    )
+                    .await {
+                    Ok((stream, latest_splits)) => Ok((stream, latest_splits)),
+                    Err(e) => {
+                        tracing::warn!(error = %e.as_report(), "failed to build source stream, retrying...");
+                        Err(e)
+                    },
+                }
+            })
+            .instrument(tracing::info_span!("build_source_stream_with_retry"))
+            .await
+            .expect("Retry build source stream until success.")
+        });
+
+        let mut need_resume_after_build = false;
+        // loop to create source stream until success
+        loop {
+            if let Some(barrier) = build_source_stream_and_poll_barrier(
```

Comment:
FYI. Link a [Slack conversation](https://risingwave-labs.slack.com/archives/C04S2HVU99V/p1732778516091719?thread_ts=1732640723.960199&cid=C04S2HVU99V) here.


> @tabVersion: It also affects the process of creating table, auto retry here means the create mv / create table command will always succeed, even if meeting some runtime error, such as Kafka group auth issue. It prevents rw from consuming data and requires the user setting up the ACL. We can only find this issue when building an actor. 

So does this issue get noticed here? What happen if I create a source with some bad settings (such the permission issue mentioned above)?

Reply:
> It prevents rw from consuming data and requires the user setting up the ACL. We can only find this issue when building an actor.

I think MQ source enumerator should employ a validation logic to check these prerequisites, so that prerequisite for connectivity should be no problem when we are creating executors. But I am not sure whether we have the check for MQ sources. cc @tabVersion

Reply:
+1 for relying on the validation logic. 

(Just saying, if someone is too lazy to write a full set of validation, they may just build a real `Consumer` and see if anything goes well.)